docs/git/branches/branches.md

```diff
@@ -0,0 +1,231 @@
+
+#### Naming
+
+Just like with variables and files, you want to name your branches in ways that will help you remember/understand what they're for. We normally name git branches with 1-5 words, with no spaces. Some good examples:
+
+- `add-story`
+- `make-button-red`
+- `delete-out-of-date-news`
+
+And some bad examples:
+
+- `branch1` or `branch2`
+- `new`
+- `feature`
+- `addalltheingredientsbutalsoremovevanilla`
+
+### Exercise 3 (10 minutes)
+
+:::tip Exercise
+Make another branch based on `main` called `extra-topping` (remember to make sure you're on the `main` branch before creating the new one!).
+
+Add a new topping to the recipe. Make sure to include it both in the ingredients list, and the steps.
+
+Commit your changes.
+:::
+
+## 4) Managing branches for pull requests
+
+We've seen before that to make a pull request from the git CLI the steps are to `git push` (to make the remote's version of `main` be the same as your local version), and then make a pull request from the GitHub UI.
+
+Most developers avoid doing work on their `main` branch, and always work on some other branch. This is because if you commit things to `main`, all new branches you create will have those changes on, and you generally want to make pull requests with only a specific set of changes compared to upstream's `main`.
+
+From now on, your steps to start working on code for a pull request should be:
+
+- `git switch main` (switch back to the main branch)
+- `git pull` (pull in any changes from upstream)
+- `git switch --create my-new-branch-name` (make a new branch)
+- (make your changes)
+- `git commit`
+- `git push` (send your changes to the remote)
+- Create a pull request through the GitHub UI
+
+If you need to make changes (e.g. because your reviewer gave you feedback), you should switch to your branch, make the changes, commit them, and push them to the remote again. The pull request will automatically get updated with the changes you committed.
+
+## 5) Deleting branches
+
+After you merge your pull request, you should delete your branch. GitHub can't change things on your computer automatically, so you need to do it yourself.
+
+You can't delete a branch you're switched to, so first, switch to `main`: `git switch main`.
+
+Then pull in any changes from upstream (this helps git to work out whether you've already merged your PR): `git pull`.
+
+And finally, delete the branch: `git branch --delete your-branch-name`.
+
+### Teacher-led example / video
+
+TODO: Make a pull request from one of your branches. Merge it. Delete the branch. Explain that it's ok to delete the branch because we know the changes are safely on `main`.
+TODO: Show `git status` showing that branches are out of sync.
+
+### Exercise 4 (15 minutes)
+
+Make pull requests for both of your branches (one with a story, one with an extra ingredient).
+
+Pair up with another trainee in your class. Get a them to review each one of your two pull requests. You should also review theirs.
```

Comment:
Maybe pull this task out to a piece of coursework? 

I've found that students put more importance on it if it's in Google Classroom as a task - I think students may skip this step unless they're already pairing